src/main/scala/com/fulcrumgenomics/commons/io/CommandLineTool.scala

```diff
@@ -0,0 +1,231 @@
+    val pipe2    = Io.pipeStream(process.getInputStream, logger.debug)
+    val exitCode = process.waitFor()
+    pipe1.close()
+    pipe2.close()
+    if (exitCode != 0) throw ToolException(exitCode)
+  }
+
+  /** Extracts a resource from the classpath and writes it to a temp file on disk.
+    *
+    * @param resource a given name on the classpath
+    * @return path to the temporary file
+    * */
+  private def writeResourceToTempFile(resource: String): Path = {
+    val lines = Io.readLinesFromResource(resource)
+    val dir   = Io.makeTempDir(PathUtil.sanitizeFileName(this.getClass.getSimpleName))
+    val path  = PathUtil.pathTo(dir.toString, PathUtil.basename(resource, trimExt = false))
+    dir.toFile.deleteOnExit()
+    Io.writeLines(path, lines)
+    path
+  }
+}
+
+/** Defines values used to get the version of the executable. */
+trait Versioned {
+  self: CommandLineTool =>
+
+  /** The default version flag. */
+  val versionFlag: String   = "--version"
+
+  /** Argument used to test version of the executable. */
+  val testArgs: Seq[String] = Seq(versionFlag)
+
+  /** Version of this executable. */
+  lazy val version: Try[ListBuffer[String]] = execCommand(executable +: testArgs: _*)
+}
+
+/** Defines methods used to check if specific modules are installed with the executable . */
+trait Modular {
+  self: CommandLineTool =>
+
+  /** Command used to test if a module is included with the tested executable. */
+  def TestModuleCommand(module: String): Seq[String]
+
+  /** Command used to test if multiple modules are included with the tested executable. */
+  def TestModuleCommand(modules: String*): Seq[Seq[String]] = modules.map(TestModuleCommand)
+
+  /** Returns true if the tested module exist with the tested executable. */
+  def ModuleAvailable(module: String): Boolean = execCommand(TestModuleCommand(module): _*).isSuccess
+
+  /** Returns true if all tested modules exist with the tested executable.
+    *
+    * For example:
+    * {{
+    * scala> import com.fulcrumgenomics.commons.io._
+    * scala> Rscript.ModuleAvailable(Seq("ggplot2", "dplyr"))
+    * res1: Boolean = true
+    * }}
+    */
+  def ModuleAvailable(modules: Seq[String], parallelism: Int = availableProcessors): Boolean = {
+    modules.parWith(parallelism).map(ModuleAvailable).forall(_ == true)
+  }
+}
+
+/** A collection of values and methods specific for the Rscript executable */
+object Rscript extends CommandLineTool with Versioned with Modular with ScriptRunner {
+  val executable: String     = "Rscript"
+  val suffix: FilenameSuffix = ".R"
+
+  /** Command to test if a R module exists.
+    *
+    * @param module name of the module to be tested
+    * @return command used to test if a given R module is installed
+    * */
+  def TestModuleCommand(module: String): Seq[String] = Seq(executable, "-e", s"stopifnot(require('$module'))")
+
+  /** True if Rscript exists. */
+  override lazy val available: Boolean = execCommand(executable, versionFlag).isSuccess
```

Comment:
I think you can safely delete this method as the overridden behavior is the same as the underlying behavior.